### PR TITLE
Small async changes to assessment request

### PR DESCRIPTION
**Pull request by tomlovesgithub**
I was unsure of where the function was failing, as there was no errors logged and the function was returning true.
I believe that these changes, specifically where noted below have fixed the issue, and testing between this branch and develop, confirms this.

### CODE REVIEW

File: functions/actions/assessments.js [hunk 1]
```diff
@@ -281,11 +281,12 @@ module.exports = (config, firebase, db) => {
...
     const exercise = await getExercise(params.exerciseId);
 
+    
     // if param is applicationId then we expect exercise to have started sending IAs. So check this.
-      // if this is not the case then we need to initialise/create the IA, send the request and update exercise stats
+    // if this is not the case then we need to initialise/create the IA, send the request and update exercise stats
 
     // get assessments
-    let assessmentsRef = db.collection('assessments')
+    let assessmentsRef = await db.collection('assessments')
```
Comment (warrensearle) [inline functions/actions/assessments.js:289]: Using an `await` here doesn't look right. We are constructing `assessmentsRef` in lines 289 - 301 and actually using it, with `await`, on line 303

**Review by warrensearle — CHANGES_REQUESTED** [review]
@tomlovesgithub can you walk me through these changes. I'm a little confused!

File: functions/actions/assessments.js [hunk 2]
```diff
@@ -298,18 +299,23 @@ module.exports = (config, firebase, db) => {
...
       assessmentsRef = assessmentsRef.where(firebase.firestore.FieldPath.documentId(), '==', params.assessmentId);
     }
-    const assessments = await getDocuments(assessmentsRef);
 
-    let result = validateAssessorEmailAddresses(assessments);
+    const assessments = await getDocuments(assessmentsRef).then((res)=>{
+      return res;
+    });
+    
+    let result = await validateAssessorEmailAddresses(assessments).then((res) => {
```
Comment (tomlovesgithub) [inline functions/actions/assessments.js:307]: here ^^

File: functions/actions/assessments.js [hunk 3]
```diff
@@ -301,6 +301,7 @@ module.exports = (config, firebase, db) => {
     const assessments = await getDocuments(assessmentsRef);
 
     let result = validateAssessorEmailAddresses(assessments);
+
```
Comment (tomlovesgithub) [inline functions/actions/assessments.js:304]: 😚👌

**Review by warrensearle — APPROVED** [review]
Thanks @tomlovesgithub this is a good fix 👍

**Review by HalcyonJAC — APPROVED** [review]

**Pull request merged**